src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/__tests__/FlagPipelineTab.test.tsx

```diff
@@ -26,117 +26,145 @@ const renderComponent = (props: Partial<FlagPipelineTabProps> = {}): void => {
+        loading: false,
+        refetch: jest.fn()
+      } as any)
     })
 
-    // select a pipeline
-    const itemToSelect = screen.getAllByRole('listitem')[4]
-    userEvent.click(itemToSelect)
-    await waitFor(() => expect(itemToSelect).toHaveClass('cardActive'))
+    test('it should create flag pipeline correctly', async () => {
+      renderComponent()
 
-    const savePipelineButton = screen.getByRole('button', { name: 'cf.featureFlags.flagPipeline.drawerButtonText' })
-    expect(savePipelineButton).toBeInTheDocument()
-  })
+      // assert empty state initially
+      const addFlagPipelineButton = screen.getByRole('button', {
+        name: 'cf.featureFlags.flagPipeline.noDataButtonText'
+      })
+      expect(screen.getByText('cf.featureFlags.flagPipeline.noDataMessage')).toBeInTheDocument()
+      expect(addFlagPipelineButton).toBeInTheDocument()
+      expect(screen.getByText('cf.featureFlags.flagPipeline.noDataDescription')).toBeInTheDocument()
 
-  test('it should let user search for a flag pipeline', async () => {
-    renderComponent()
-    const addFlagPipelineButton = screen.getByRole('button', {
-      name: 'cf.featureFlags.flagPipeline.noDataButtonText'
-    })
-    userEvent.click(addFlagPipelineButton)
-
-    const searchInput = screen.getByPlaceholderText('Search')
-    expect(searchInput).toBeInTheDocument()
-
-    userEvent.type(searchInput, 'pipeline 1', { allAtOnce: true })
-
-    await waitFor(() =>
-      expect(refetchMock).toBeCalledWith({
-        debounce: 500,
-        queryParams: {
-          accountIdentifier: 'dummy',
-          environmentIdentifier: '',
-          identifier: 'TEST_FLAG',
-          orgIdentifier: 'dummy',
-          pageNumber: 0,
-          pageSize: 100,
-          pipelineName: 'pipeline 1',
-          projectIdentifier: 'dummy'
-        }
+      // assert drawer opens with correct text. Save button hidden initially
+      userEvent.click(addFlagPipelineButton)
+      await waitFor(() => {
+        expect(screen.getByText('cf.featureFlags.flagPipeline.drawerTitle')).toBeInTheDocument()
+        expect(screen.getByText('cf.featureFlags.flagPipeline.drawerDescription')).toBeInTheDocument()
+        expect(screen.getByText('cf.featureFlags.flagPipeline.envText')).toBeInTheDocument()
+        expect(
+          screen.queryByRole('button', { name: 'cf.featureFlags.flagPipeline.drawerButtonText' })
+        ).not.toBeInTheDocument()
       })
-    )
-  })
 
-  test('it should close drawer when close icon clicked', async () => {
-    renderComponent()
-    const addFlagPipelineButton = screen.getByRole('button', {
-      name: 'cf.featureFlags.flagPipeline.noDataButtonText'
-    })
+      // assert correct number of pipelines and pipeline content is displayed
+      await waitFor(() => {
+        expect(screen.getAllByRole('listitem')).toHaveLength(mockAvailablePipelines.length)
+        expect(screen.getAllByRole('listitem')[0]).toHaveTextContent('Pipeline 1Pipeline 1 description')
```

Comment:
Super-minor optimisation... you can save an operation by extracting `screen.getAllByRole('listitem')` to a variable. Otherwise it gets run twice.
```suggestion
        const items = screen.getAllByRole('listitem')
        expect(items).toHaveLength(mockAvailablePipelines.length)
        expect(items[0]).toHaveTextContent('Pipeline 1Pipeline 1 description')
```

Reply:
fixed